src/Microsoft.Health.Fhir.CosmosDb/Features/Search/Queries/QueryBuilder.cs

```diff
@@ -129,6 +134,41 @@ public QueryDefinition GenerateHistorySql(SearchOptions searchOptions)
                 return query;
             }
 
+            public QueryDefinition GenerateReindexSql(SearchOptions searchOptions, string searchParameterHash)
+            {
+                EnsureArg.IsNotNull(searchOptions, nameof(searchOptions));
+
+                if (searchOptions.CountOnly)
+                {
+                    AppendSelectFromRoot("VALUE COUNT(1)");
+                }
+                else
+                {
+                    AppendSelectFromRoot();
+                }
+
+                AppendSystemDataFilter();
+
+                var expressionQueryBuilder = new ExpressionQueryBuilder(
+                    _queryBuilder,
+                    _queryParameterManager);
+
+                if (searchOptions.Expression != null)
+                {
+                    _queryBuilder.Append("AND ");
+                    searchOptions.Expression.AcceptVisitor(expressionQueryBuilder);
+                }
+
+                AppendFilterCondition(
```

Comment:
Should we also filter for `IsDeleted` and `IsHistory` in this query?

Reply:
I want to include the history, as we need to reindex the history as well, but I did miss the IsDeleted.  Thanks!